src/Compilers/CSharp/Test/Symbol/Compilation/GetImportScopesTests.cs

```diff
@@ -0,0 +1,753 @@
+            Assert.Empty(scopes);
+        }
+
+        [Fact]
+        public void TestBeforeImportsTopLevelStatements()
+        {
+            var text = @"
+/*pos*/
+using System;
+
+return;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Single(scopes);
+            Assert.Single(scopes.Single().Imports);
+            Assert.True(scopes.Single().Imports.Single().NamespaceOrType is INamespaceSymbol { ContainingNamespace.IsGlobalNamespace: true, Name: nameof(System) });
+            Assert.True(scopes.Single().Imports.Single().DeclaringSyntaxReference!.GetSyntax() is UsingDirectiveSyntax);
+            Assert.Empty(scopes.Single().Aliases);
+            Assert.Empty(scopes.Single().ExternAliases);
+            Assert.Empty(scopes.Single().XmlNamespaces);
+        }
+
+        [Fact]
+        public void TestAfterImportsTopLevelStatements1()
+        {
+            var text = @"
+using System;
+/*pos*/
+return;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Empty(scopes);
+        }
+
+        [Fact]
+        public void TestAfterImportsTopLevelStatements2()
+        {
+            var text = @"
+using System;
+
+return /*pos*/;";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Single(scopes);
+            Assert.Single(scopes.Single().Imports);
+            Assert.True(scopes.Single().Imports.Single().NamespaceOrType is INamespaceSymbol { ContainingNamespace.IsGlobalNamespace: true, Name: nameof(System) });
+            Assert.True(scopes.Single().Imports.Single().DeclaringSyntaxReference!.GetSyntax() is UsingDirectiveSyntax);
+            Assert.Empty(scopes.Single().Aliases);
+            Assert.Empty(scopes.Single().ExternAliases);
+            Assert.Empty(scopes.Single().XmlNamespaces);
+        }
+
+        [Fact]
+        public void TestAfterMultipleImportsNoContent()
+        {
+            var text = @"
+using System;
+using Microsoft;
+/*pos*/";
+            var tree = Parse(text);
+            var comp = CreateCompilation(tree);
+            var model = comp.GetSemanticModel(tree);
+            var scopes = model.GetImportScopes(GetPositionForBinding(text));
+            Assert.Single(scopes);
+            Assert.Equal(2, scopes.Single().Imports.Length);
+            Assert.True(scopes.Single().Imports.First().NamespaceOrType is INamespaceSymbol { ContainingNamespace.IsGlobalNamespace: true, Name: nameof(System) });
```

Comment:
Although the API explicitly doesn't commit to ordering entries within a scope, we're testing as if we were committed to that order. If we decide to stick with that API, we should consider using `ConditionallyDeOrder` (and update tests accordingly)

It may be worth extracting a method to verify scopes textually (dump scopes to text in some format, then compare to expected string).

Reply:
actually, this test is written poorly, i uipdated the rest to use .Any, and that's what i should be doing here.

Reply:
It looks like folks agreed on keeping an unordered API. I think it would be a good idea to add `ConditionallyDeOrder` as we do for some other explicitly unordered APIs.
Note this may affect tests that still assume a certain order.

Reply:
> It looks like folks agreed on keeping an unordered API. I think it would be a good idea to add ConditionallyDeOrder as we do for some other explicitly unordered APIs.

Done.  THanks!